File: app/src/main/java/com/vsa/filmoteca/presenter/detail/DetailPresenter.java
package com.vsa.filmoteca.presenter.detail;

import android.support.v4.widget.SwipeRefreshLayout;

/**
 * Created by seldon on 13/03/15.
 */
public interface DetailPresenter extends SwipeRefreshLayout.OnRefreshListener {

    void onCreate(String url, String movieTitle);

    void onDestroy();

    void onRefresh();

    void onShareButtonClick();

    void onShowInBrowserButtonClick();

    void onFilmAffinitySearchButtonClick();

    void onAboutUsButtonClick();

    void onFabClick();

}

File: app/src/main/java/com/vsa/filmoteca/presentation/presenter/detail/DetailPresenter.kt
package com.vsa.filmoteca.presentation.presenter.detail

/**
 * Created by Alberto Vecina Sánchez on 2019-05-08.
 */
interface DetailPresenter {

    fun onCreate(url: String, movieTitle: String)
    
    fun onShareButtonClick()

    fun onShowInBrowserButtonClick()

    fun onFilmAffinitySearchButtonClick()

    fun onRefresh()

    fun onAboutUsButtonClick()

    fun onFabClick()

}